### ISSUE TITLE
Mac OS support

### ISSUE THREAD

**Issue by mahesh-hegde** [dart-lang/native#780]
I don't have a Mac OS setup. Therefore unable to build dartjni for macos on desktop.

It's build system is different (podspec), and it may take more effort than porting same CMake config to windows.

I searched through github and sourcegraph, and unfortunately not able to find a single podspec that makes use of JNI libraries.

So this would involve finding JNI libraries and listing them in podspec then fixing os-specific errors / warnings..

edit @dcharkes list of tasks that we should do

- [x] run package:jni tests on MacOS (on CI)
- [x] run package:jnigen tests on MacOS (on CI)
- [x] MacOS CLI apps using some jnigen'ed code working (on CI)
- [x] MacOS Flutter desktop app using some jnigen'ed code working
- [x] MacOS as host for running jnigen on some Java sources (on CI)
- [ ] MacOS as host for Android Flutter app working

Not all of these can be run on the CI unfortunately (e.g. Flutter apps can be build, but not run).

**Comment by HosseinYousefi** [comment]
Not a priority and blocked by https://github.com/flutter/flutter/issues/152221. Moving to backlog.
